docs/assessment.md

```diff
@@ -128,6 +128,82 @@ By default, PrairieLearn provides an incentive for students to answer a specific
+
+The basic idea is that students should receive `autoPoints` each time they solve the question successfully, up to a maximum of `maxAutoPoints`. When partial credit is awarded, a student receives partial points for the question, but only if they are getting closer to a correct answer. For example, if a student repeatedly answers a question with 50% correct answers then they will receive 50% points for the first such answer, but they will not be able to build up points to reach `maxAutoPoints`. They will need to answer the question 100% correctly to be able to advance their points past the 50% mark.
+
+The exact details of how PrairieLearn awards points are as follows. First, a number of variables are computed internally by PrairieLearn as the student answers the question:
+- **Submission percentage score** is the score (between 0% and 100%) that the student receives for any given submission attempt, as evaluated by the question grading specification.
+- **Best current score** is the highest submission percentage score that the student has received since last achieving 100% for a submission. This is initialized to zero.
+- **Question value** is the current maximum number of points that the question is worth. This starts at `autoPoints` and increases by `autoPoints` for each consecutive correct submission. This is initialized to `autoPoints`.
+- **Question points** is the total number of points earned by the student for the question. This will never decrease. This is initialized to zero.
+
+For each submission attempt made by a student, the scoring algorithm is as follows:
+1. If the `submission percentage score` is more than `best current score`, add `(submission percentage score - best current score) / 100 * question value` to `question points`, capping `question points` at `maxAutoPoints`.
+2. If the `submission percentage score` is 100%, then reset `best current score` to zero. Otherwise, if the `submission percentage score` is more than `best current score`, then set `best current score` to `submission percentage score`.
+3. If the `submission percentage score` is 100%, then increase `question value` by `autoPoints`. Otherwise set `question value` to `autoPoints`. If `constantQuestionValue` is true then skip this step.
+
+As an example, suppose a student makes 8 submissions of varying degrees of correctness to a question. Assume the question has `autoPoints` set to 4 and `maxAutoPoints` set to 10. The student's submissions are as follows:
+
+Submission number | Submission perc. score | Best current score | Question value | Question points | Comment
+--- | --- | --- | --- | --- | ---
+Init | - | 0 | 4 | 0 | Initial values
+1 | 50% | 50% | 4 | 2 | (50% - 0%) * 4 = 2 points awarded
+2 | 80% | 80% | 4 | 3.2 | (80% - 50%) * 4 = 1.2 points awarded
+3 | 20% | 80% | 4 | 3.2 | No improvement in score, so no points awarded
```

Comment:
```suggestion
3 | 20% | 80% | 4 | 3.2 | No improvement in score, so no points awarded or deducted
```

Reply:
Done in 4191a4335164ac391314b4b9f76d725eba219f17